File: backend/src/main/java/net/lycee/web/enquete/api/entity/QuestionEntity.java
package net.lycee.web.enquete.api.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import net.lycee.web.enquete.api.domain.QuestionId;
import net.lycee.web.enquete.api.domain.QuestionType;
import net.lycee.web.enquete.api.domain.SpaceId;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class QuestionEntity {
    private QuestionId questionId;
    private SpaceId spaceId;
    private Integer no;
    private QuestionType type;
    private String description;
    private Long endTime;

    private List<AnswerEntity> answers;
}

File: frontend/src/types/QuestionEntity.ts
import type { QuestionId, QuestionType } from "./common.ts";
import type { AnswerEntity } from "./AnswerEntity.ts";

export type QuestionEntity = {
  // 質問ID
  id: QuestionId;

  // 質問タイプ
  type: QuestionType;

  // 質問内容
  content: string;

  // 回答群
  answers: AnswerEntity[];

  // 複数回答フラグ
  isMultiple: boolean;

  // 回答期限
  closeTime: Date;
};
